pallets/rmrk-core/src/functions.rs

```diff
@@ -69,6 +71,41 @@ where
 		Properties::<T>::insert((&collection_id, maybe_nft_id, &key), &value);
 		Ok(())
 	}
+
+	// Internal function to set a property for downstream `Origin::root()` calls.
+	fn do_set_property(
+		origin: T::Origin,
+		collection_id: CollectionId,
+		maybe_nft_id: Option<NftId>,
+		key: KeyLimitOf<T>,
+		value: ValueLimitOf<T>,
+	) -> sp_runtime::DispatchResult {
+		// Check if root user
+		ensure_root(origin)?;
```

Comment:
Since this is just internal API for other pallets, it's not really necessary to pass an additional `origin` argument. After all, whenever a pallet wants to call this function, they just pass `Origin::root()`, which is equivalent to nothing.

This function can also be called from the user facing dispatchables. It's the caller's duty to check the permission on their own.

Reply:
Updated